### DISCUSSION TITLE
Pre-auth before uploading using SSH/TLS based transfer

### DISCUSSION THREAD

**Discussion by Avi-Roy1** [curl/curl#11193] [Q&A]
Hello community members,

I need to pre-authenticate/authorize a server before uploading a file using libcurl. The issue is it takes a lot of time to generate the file and if there is inadequate directory permission or wrong credentials we would know only after a while. So could you suggest any possible way to do this?

**Comment by bagder** [comment]
Upload a tiny dummy file first to check that it works?

**Reply by Avi-Roy1** [reply]
Yes, that is my current approach, but I would like to know if there is any alternative. 
As for the current approach, I have to upload a dummy file, check for errors, and delete that file from the fileserver.
